trin-state/src/network.rs

```diff
@@ -35,7 +35,8 @@ impl StateNetwork {
         // Trigger bonding with bootnodes, at both the base layer and portal overlay.
         // The overlay ping via talkreq will trigger a session at the base layer, then
         // a session on the (overlay) portal network.
-        for enr in rw_read!(self.overlay.discovery).discv5.table_entries_enr() {
+        let table_entries = rw_read!(self.overlay.discovery).discv5.table_entries_enr();
+        for enr in table_entries {
```

Comment:
So I take it that the old lock was being held for the whole iteration, and now it's not?

Reply:
Yeah, that's my impression as well. But I don't have any special knowledge as to how rust works, I made a guess that this would cause Rust to drop the lock sooner and it seems to have worked!